pages/desktop/base.py

```diff
@@ -183,6 +202,156 @@ def click_logout(self):
         action.perform()
         self.wait.until(lambda s: self.is_element_displayed(*self._login_locator))
 
+    def click_user_menu_collections_link(self):
+        user = WebDriverWait(
+            self.selenium, 30, ignored_exceptions=StaleElementReferenceException
+        ).until(
+            EC.element_to_be_clickable(
+                (
+                    By.CSS_SELECTOR,
+                    '.Header-user-and-external-links \
+            .DropdownMenu-button-text',
+                )
+            )
+        )
+        link = self.find_element(*self._user_menu_collection_link_locator)
+        dropdown = self.find_element(*self._account_dropdown_locator)
+        loop = 0
+        while loop < 3:
```

Comment:
While using ActionChains to interact with elements from the user dropdown menu, I've received the following intermittent errors:
- ` StaleElementReferenceException` mostly for the `user` element (for which I've already added a wait condition that was supposed to ignore that exception but it didn't prove enough)
- `WebDriverException: Message: TypeError: rect is undefined` - which, after some research, seems to be a pretty common error encountered only in Firefox - https://github.com/mozilla/geckodriver/issues/1269

I've tried to add more waits to the other elements I'm interacting with but the errors were still occurring.  The solution I've ended up implementing is a while loop with a `try - except` method which tries to catch these exceptions and retry the action chains up to 3 times, if they occur.

I had to use the same method for all the other elements from the user dropdown menu.

Reply:
Do you think we can put those in their own fixtures and just return the ```user```?

Reply:
I could try, but I'm not sure if the exceptions occur only for the `user`.

Reply:
Actually, I'm not so sure how to do this

Reply:
I went for a more simple solution that avoids code repetition: I've added one single function (instead of four) that works for all user menu links which asks for two parameters that can be passed in the individual tests, namely a `count` that specifies which link to click and a `landing_page` that contains the element we want to find on the page being opened.